From de8d37db42fb4c151d42f4c4b72d73a73be29cd1 Mon Sep 17 00:00:00 2001
From: Maschell <Maschell@gmx.de>
Date: Sat, 18 Feb 2023 09:43:46 +0100
Subject: [PATCH] coreinit: Fix OSDynLoad_Error to be a 32bit value

---
 include/coreinit/dynload.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/coreinit/dynload.h b/include/coreinit/dynload.h
index f17631d..e4ae853 100644
--- a/include/coreinit/dynload.h
+++ b/include/coreinit/dynload.h
@@ -33,7 +33,7 @@ typedef enum OSDynLoad_Error
    OS_DYNLOAD_INVALID_ALLOCATOR_PTR       = 0xBAD10017,
    OS_DYNLOAD_OUT_OF_SYSTEM_MEMORY        = 0xBAD1002F,
    OS_DYNLOAD_TLS_ALLOCATOR_LOCKED        = 0xBAD10031,
-   OS_DYNLOAD_MODULE_NOT_FOUND            = -6,
+   OS_DYNLOAD_MODULE_NOT_FOUND            = 0xFFFFFFFA,
 } OSDynLoad_Error;
 
 typedef OSDynLoad_Error (*OSDynLoadAllocFn)(int32_t size, int32_t align, void **outAddr);
-- 
2.39.2.windows.1

