diff --git a/music_ogg.c b/music_ogg.c
index 8323653..270eadf 100644
--- a/music_ogg.c
+++ b/music_ogg.c
@@ -209,7 +209,7 @@ static int OGG_UpdateSection(OGG_music *music)
         music->stream = NULL;
     }
 
-    music->stream = SDL_NewAudioStream(AUDIO_S16, vi->channels, (int)vi->rate,
+    music->stream = SDL_NewAudioStream(AUDIO_S16SYS, vi->channels, (int)vi->rate,
                                        music_spec.format, music_spec.channels, music_spec.freq);
     if (!music->stream) {
         return -1;
@@ -343,7 +343,7 @@ static int OGG_GetSome(void *context, void *data, int bytes, SDL_bool *done)
 #ifdef OGG_USE_TREMOR
     amount = vorbis.ov_read(&music->vf, music->buffer, music->buffer_size, &section);
 #else
-    amount = (int)vorbis.ov_read(&music->vf, music->buffer, music->buffer_size, 0, 2, 1, &section);
+    amount = (int)vorbis.ov_read(&music->vf, music->buffer, music->buffer_size, SDL_BYTEORDER == SDL_BIG_ENDIAN, 2, 1, &section);
 #endif
     if (amount < 0) {
         set_ov_error("ov_read", amount);
